flutter disturb the output for gradle command : ./gradlew tasks

## Steps to Reproduce
1.add flutter to android project use gradle.
2.run `./gradlew tasks` to print gradle tasks.

**Expected results:** 
print gradle tasks only.

**Actual results:** <!-- what did you see? -->
print :dependencies for all flavors firstly, and print gradle tasks behind.

<p>

#### output for `flutter doctor -v`:
```
[✓] Flutter (Channel stable, 2.5.1, on macOS 12.0 21A5506j darwin-x64, locale zh-Hans-CN)
    • Flutter version 2.5.1 at /usr/local/libs/flutter_sdk
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ffb2ecea52 (4 天前), 2021-09-17 15:26:33 -0400
    • Engine revision b3af521a05
    • Dart version 2.14.2
    • Pub download mirror https://pub.flutter-io.cn
    • Flutter download mirror https://storage.flutter-io.cn

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/yaocf/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Users/yaocf/Library/Android/sdk
    • ANDROID_SDK_ROOT = /Users/yaocf/Library/Android/sdk
    • Java binary at: /Applications/Android Studio 4.2 Preview.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)
    • All Android licenses accepted.

[!] Xcode - develop for iOS and macOS
    ✗ Xcode installation is incomplete; a full installation is necessary for iOS development.
      Download at: https://developer.apple.com/xcode/download/
      Or install Xcode via the App Store.
      Once installed, run:
        sudo xcode-select --switch /Applications/Xcode.app/Contents/Developer
        sudo xcodebuild -runFirstLaunch
    • CocoaPods version 1.10.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio 4.2 Preview.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

[✓] Connected device (1 available)
    • Chrome (web) • chrome • web-javascript • Google Chrome 93.0.4577.82

! Doctor found issues in 1 category.
```

</p>

Hi @chunfengyao 
I just tried your module with `flutter.gradle` modfiied 
```console
PS C:\Users\Taha\StudioProjects\issueSimpleProject\module_flutter\.android> .\gradlew tasks

> Task :tasks FAILED

------------------------------------------------------------
Tasks runnable from root project
------------------------------------------------------------

generateLockfiles

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tasks'.
> Could not create task ':generateLockfiles'.
   > A problem occurred starting process 'command 'echo''

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.7/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 1s
1 actionable task: 1 executed
PS C:\Users\Taha\StudioProjects\issueSimpleProject\module_flutter\.android>
```

> Hi @chunfengyao
> I just tried your module with `flutter.gradle` modfiied
> 
> ```
> PS C:\Users\Taha\StudioProjects\issueSimpleProject\module_flutter\.android> .\gradlew tasks
> 
> > Task :tasks FAILED
> 
> ------------------------------------------------------------
> Tasks runnable from root project
> ------------------------------------------------------------
> 
> generateLockfiles
> 
> FAILURE: Build failed with an exception.
> 
> * What went wrong:
> Execution failed for task ':tasks'.
> > Could not create task ':generateLockfiles'.
>    > A problem occurred starting process 'command 'echo''
> 
> * Try:
> Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
> 
> * Get more help at https://help.gradle.org
> 
> Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
> Use '--warning-mode all' to show the individual deprecation warnings.
> See https://docs.gradle.org/6.7/userguide/command_line_interface.html#sec:command_line_warnings
> 
> BUILD FAILED in 1s
> 1 actionable task: 1 executed
> PS C:\Users\Taha\StudioProjects\issueSimpleProject\module_flutter\.android>
> ```

thanks @TahaTesser 
you can try this project directly (without modified the flutter.gradle), I modified it is only for reduce the console output.
and, you can see from your output , when you run `./gradlew tasks` , it has tried to run `generateLockfiles` task (so ,you saw the message `A problem occurred starting process 'command 'echo''`) . actually the `./gradlew tasks` command should only print gradle tasks but should'n run any of them.

--- a/packages/flutter_tools/gradle/flutter.gradle
+++ b/packages/flutter_tools/gradle/flutter.gradle
@@ -103,14 +103,16 @@ class FlutterPlugin implements Plugin<Project> {
         this.project = project
 
         def rootProject = project.rootProject
-        rootProject.tasks.register('generateLockfiles') {
-            rootProject.subprojects.each { subproject ->
-                def gradlew = (OperatingSystem.current().isWindows()) ?
-                    "${rootProject.projectDir}/gradlew.bat" : "${rootProject.projectDir}/gradlew"
-                rootProject.exec {
-                    workingDir rootProject.projectDir
-                    executable gradlew
-                    args ":${subproject.name}:dependencies", "--write-locks"
+        if (isFlutterAppProject()) {
+            rootProject.tasks.register('generateLockfiles') {
+                rootProject.subprojects.each { subproject ->
+                    def gradlew = (OperatingSystem.current().isWindows()) ?
+                        "${rootProject.projectDir}/gradlew.bat" : "${rootProject.projectDir}/gradlew"
+                    rootProject.exec {
+                        workingDir rootProject.projectDir
+                        executable gradlew
+                        args ":${subproject.name}:dependencies", "--write-locks"
+                    }
                 }
             }
         }

